src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs

```diff
@@ -0,0 +1,203 @@
+                { "websdk.b55d4f4cf22bee7ec9a2ca5f49d54ebf6ee67e83",                  "A51DEB15209039D17FC0D781BFC445F8C5CDC1D51673AC4C0929FBEE8C1E4D21" },
+                { "winforms.b1ee29b8b8e14c1200adff02847391dde471d0d2",                "1288479D12130F9AEF56137DCC655B5B59277F5153172B30DC419456AF9CA011" },
+                {"wpf.d378b1ec6b8555c52b7da1c40ffc0784cb0f5cad",                      "CFE5E19366FD5A90101A8F369CFC70611F9B1439A0720D435EE34872AF55A40A" }
+            };
+
+            string[] expectedMasterRepos2 = new string[]
+            {
+                "cli",
+                "cliCommandLineParser",
+                "core-sdk",
+                "coreclr",
+                "msbuild",
+                "standard",
+                "toolset",
+                "websdk",
+                "winforms",
+                "wpf"
+            };
+
+            string[] expectedGitDirs2 = new string[]
+            {
+                "cli.git",
+                "cliCommandLineParser.git",
+                "core-sdk.git",
+                "coreclr.git",
+                "msbuild.git",
+                "standard.git",
+                "toolset.git",
+                "websdk.git",
+                "winforms.git",
+                "wpf.git"
+            };
+
+            reposToIgnore += ";https://github.com/dotnet/arcade";
+            TestContext.WriteLine($"Cloning repo {sourceRepoUri} at {sourceRepoVersion} with depth 4 and include-toolset=true");
+            reposFolder = await CloneRepositoryWithDarc(sourceRepoName, sourceRepoVersion, reposToIgnore, true, 4);
+            CheckExpectedClonedRepos(expectedRepos2, expectedMasterRepos2, expectedGitDirs2, reposFolder);
+        }
+
+        private void CheckExpectedClonedRepos(Dictionary<string, string> expectedRepos, string[] expectedMasterRepos, string[] expectedGitDirs, TemporaryDirectory reposFolder)
+        {
+            using (ChangeDirectory(reposFolder.Directory))
+            {
+                string clonedReposFolder = Path.Join(reposFolder.Directory, "cloned-repos");
+                string gitDirFolder = Path.Join(reposFolder.Directory, "git-dirs");
+
+                TestContext.WriteLine("Validate the hash of the Version.Details.xml matches expected");
+                foreach (string name in expectedRepos.Keys)
+                {
+                    string path = Path.Join(clonedReposFolder, name);
+                    DirectoryAssert.Exists(path, $"Expected cloned repo '{name}' but not found at {path}");
+
+                    string versionPath = Path.Join(path, "eng", "Version.Details.xml");
+                    FileAssert.Exists(versionPath, $"Expected a file at {versionPath}");
+
+                    using (FileStream stream = File.OpenRead(versionPath))
+                    {
+                        SHA256 hashGenerator = SHA256.Create();
```

Comment:
This is disposable too
```suggestion
                        using SHA256 hashGenerator = SHA256.Create();
```